spec/lib/payment_calculator/ecf/contract/uplift_payment_calculations_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require "payment_calculator/ecf/contract/uplift_payment_calculations"
+
+class DummyClass
+  include PaymentCalculator::Ecf::Contract::UpliftPaymentCalculations
+end
+
+describe ::PaymentCalculator::Ecf::Contract::UpliftPaymentCalculations do
+  it "returns the expected types for uplift" do
+    contract = double("Contract Double", uplift_amount: BigDecimal(100.00, 2))
+    call_off_contract = DummyClass.new({ contract: contract })
+    event_type = :started
+
+    if @combined_results.nil?
+      expect(call_off_contract.uplift_payment_per_participant.round(2)).to be_a(BigDecimal)
+      expect(call_off_contract.uplift_payment_per_participant_for_event(event_type: event_type)).to be_a(BigDecimal)
+      expect(call_off_contract.uplift_payment_for_event(event_type: event_type, uplift_participants: 100)).to be_a(BigDecimal)
+    end
+  end
+
+  it "performs calculations of the uplift payments" do
+    contract = double("Contract Double", uplift_amount: 100)
+    call_off_contract = DummyClass.new({ contract: contract })
+    total_participants = 330
+
+    expect(call_off_contract.uplift_payment_per_participant.round(0)).to eq(100.00)
+
+    %i[started].each do |event_type|
+      expect(call_off_contract.uplift_payment_per_participant_for_event(event_type: event_type).round(2)).to eq(100.00)
+      expect(call_off_contract.uplift_payment_for_event(event_type: event_type, uplift_participants: total_participants).round(2)).to eq(33_000.00)
```

Comment:
We are doing turnip features for calculation outputs, I don't see a diff on them in this PR but you have implemented the calculations. I think we need them before this can be merged.

Reply:
The features were the wrong shape for the calculations required and are adding less and less value and requiring more and more maintenance. I'd suggest that we drop them since the same tests are being run in different ways and the more dynamic rake tasks are being used to generate output in the correct format.

Reply:
I'm hesitant to drop the business readable specs, but equally I'm not keen on cargo-culting it if it's just work for no value. What do you think?

Reply:
It takes longer to change the features that any other part of the coding. Adds overhead for no real value. It's also one reason that extracting the DIParameters has become so tricky since it's expecting a baseline to be created which it then calls with the totals. (Maybe that's a justification as to why we still need the new().call() structure, but I don't think it should be the only reason.)

Reply:
Done in the aggregation PR, so not required here too.

Reply:
It will be done in my branch